catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsLoader.java

```diff
@@ -288,38 +334,25 @@ private static boolean anyNull(Object... args) {
   }
 
   /**
-   * Results from this method call should not be directly returned to clients or cached in some
-   * intermediate block of memory.
+   * Caution should be used with this, as it elevates the permissions to the System user.
+   *
+   * <p>Operations that throw an {@link Exception} will be tried (in total) a maximum of 3 times
+   * before actually failing.
+   *
+   * @param func What to execute as the System
+   * @param <T> Generic return type of func
+   * @return result of the callable func
    */
-  private static Set<String> queryAsAdmin(
-      EndpointUtil util, String tag, Function<Map<String, Result>, Set<String>> transform) {
-    return SECURITY.runAsAdmin(
-        () -> {
-          try {
-            return SECURITY.runWithSubjectOrElevate(
-                () -> transform.apply(util.getMetacardsByFilter(tag)));
-          } catch (SecurityServiceException | InvocationTargetException e) {
-            LOGGER.warn(
-                "Can't query the catalog while trying to initialize system search templates, was "
-                    + "unable to elevate privileges",
-                e);
-          }
-          return Collections.emptySet();
-        });
-  }
-
-  private static void saveMetacards(CatalogFramework framework, List<Metacard> metacards) {
-    SECURITY.runAsAdmin(
-        () -> {
-          try {
-            return SECURITY.runWithSubjectOrElevate(
-                () -> framework.create(new CreateRequestImpl(metacards)).getCreatedMetacards());
-          } catch (SecurityServiceException | InvocationTargetException e) {
-            LOGGER.warn(
-                "Can't create metacard for system search template, was unable to elevate privileges",
-                e);
-          }
-          return null;
-        });
+  private static <T> T executeAsSystem(Callable<T> func) {
+    Subject systemSubject = SECURITY.runAsAdmin(SECURITY::getSystemSubject);
+    if (systemSubject == null) {
+      throw new SecurityException("Could not get systemSubject to initialize system templates");
+    }
+    return Failsafe.with(
+            new RetryPolicy()
+                .retryOn(Collections.singletonList(Exception.class))
+                .withMaxRetries(2)
+                .withBackoff(2, 10, TimeUnit.SECONDS))
```

Comment:
Good point with regard to the max timeout. 

Regarding why we do this, it's because this particular private method is only used for operations during bundle start, which includes querying and persistence. Both benefit from retry semantics in case a `SourceUnavailableException` is thrown, or during persistence if Solr hiccups. It's rare but it can happen.

Reply:
doesn't maxRetries 2 mean we try and then can retry twice, and with the backoff doesn't that mean we will go 2 originally, 4 on the first retry, and 8 on the second?  

I guess, what I am curious about, what is the behavior where this never succeeds?